controllers/istio_controller.go

```diff
@@ -103,26 +113,65 @@ func (r *IstioReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	// error occurs in the reconciliation of the Istio upgrade after the Istio upgrade.
 	proxyErr := r.proxySidecars.Reconcile(ctx, istioCR)
 	if proxyErr != nil {
-		r.log.Error(proxyErr, "Error occurred during reconciliation of Istio Sidecars")
-		return r.statusHandler.SetError(ctx, described_errors.NewDescribedError(proxyErr, "Error occurred during reconciliation of Istio Sidecars"), r.Client, &istioCR, metav1.Condition{}, ErrorRetryTime)
+		describedErr := described_errors.NewDescribedError(proxyErr, "Error occurred during reconciliation of Istio Sidecars")
+		return r.cancelReconcile(ctx, istioCR, describedErr)
 	}
 
-	// Put applied configuration in annotation
-	istioCR, updateErr := istio.UpdateLastAppliedConfiguration(istioCR, IstioTag)
-	if updateErr != nil {
-		r.log.Error(updateErr, "Error updating LastAppliedConfiguration")
-		return r.statusHandler.SetError(ctx, described_errors.NewDescribedError(updateErr, "Error updating LastAppliedConfiguration"), r.Client, &istioCR, metav1.Condition{}, ErrorRetryTime)
+	return r.finishReconcile(ctx, istioCR, IstioTag)
+}
+
+// cancelReconcile cancels the reconciliation and requeues the request.
+func (r *IstioReconciler) cancelReconcile(ctx context.Context, istioCR operatorv1alpha1.Istio, err described_errors.DescribedError) (ctrl.Result, error) {
```

Comment:
I created different functions to make it to make it easy to spot that those are different actions. Also the logic inside will get harder to understand, since there is a different handling for the result in successful and status error case and different logging. I'm not a fan of mixing this up, because this are completely different processes and we should make it obvious.